mypy/checkexpr.py

```diff
@@ -1659,21 +1668,29 @@ def check_callable_call(
         callable_node: Expression | None,
         callable_name: str | None,
         object_type: Type | None,
+        is_overload_item: bool = False,
     ) -> tuple[Type, Type]:
         """Type check a call that targets a callable value.
 
         See the docstring of check_call for more information.
         """
+        # Check implicit calls to deprecated class constructors.
+        # Only the non-overload case is handled here. Overloaded constructors are handled
+        # separately during overload resolution.
+        if (not is_overload_item) and callee.is_type_obj():
+            self.chk.warn_deprecated(callee.definition, context)
```

Comment:
This looks very nice now!

Maybe a little bit more readable: `callee.is_type_obj() and not is_overloaded_item`.